sdk/src/main/java/io/customer/sdk/di/CustomerIOComponent.kt

```diff
@@ -59,27 +41,48 @@ class CustomerIOComponent(
         get() = override() ?: QueueStorageImpl(sdkConfig, fileStorage, jsonAdapter, logger)
 
     val queueRunner: QueueRunner
-        get() = override() ?: QueueRunnerImpl(jsonAdapter, cioHttpClient, logger)
+        get() = override() ?: QueueRunnerImpl(jsonAdapter, cioHttpClient, logger, hooksManager)
 
     val queue: Queue
         get() = override() ?: QueueImpl.getInstanceOrCreate {
-            QueueImpl(dispatcher = Dispatchers.IO, queueStorage, queueRunRequest, jsonAdapter, sdkConfig, timer, logger, dateUtil)
+            QueueImpl(
+                dispatcher = Dispatchers.IO,
+                queueStorage,
+                queueRunRequest,
+                jsonAdapter,
+                sdkConfig,
+                timer,
+                logger,
+                dateUtil
+            )
         }
 
     val queueQueryRunner: QueueQueryRunner
         get() = override() ?: QueueQueryRunnerImpl(logger)
 
     val queueRunRequest: QueueRunRequest
-        get() = override() ?: QueueRunRequestImpl(queueRunner, queueStorage, logger, queueQueryRunner)
+        get() = override() ?: QueueRunRequestImpl(
+            queueRunner,
+            queueStorage,
+            logger,
+            queueQueryRunner
+        )
 
     val logger: Logger
         get() = override() ?: LogcatLogger(sdkConfig)
 
+    val hooksManager: HooksManager by lazy { override() ?: CioHooksManager() }
```

Comment:
For consistency, instead of `lazy`, should we use `getInstanceOrCreate()` like we do with the `Queue` class?